src/readonly.jl

```diff
@@ -35,5 +35,8 @@ Base.copy(x::ReadOnly) = ReadOnly(copy(parent(x)))
 (==)(x::ReadOnly, y::AbstractVector) = parent(x) == y
 (==)(x::AbstractVector, y::ReadOnly) = x == parent(y)
 (==)(x::ReadOnly, y::ReadOnly) = parent(x) == parent(y)
+# disambiguation
+(==)(x::ReadOnly{T,1,<:AbstractVector{T}}, y::ReadOnly{S,1,<:AbstractVector{S}}) where {T,S} =
```

Comment:
Why does this help? i really don't understand.

Reply:
```suggestion
(==)(x::ReadOnly{T,N,<:AbstractVector{T}}, y::ReadOnly{S,N,<:AbstractVector{S}}) where {T,S,N} =
```

Reply:
or does it make it ambiguous ?

Reply:
I'm not 100% sure TBH. But looking at

```julia
julia> typeintersect(SparseArrays.ReadOnly, AbstractVector)
SparseArrays.ReadOnly{T, 1, V} where {T, V<:AbstractVector{T}}
```

I believe the ambiguity is real for `ReadOnly` of vectors? I think we should include it for now and then turn on ambiguity testing and try to remove this method and see what happens. But with the confusion of sysimg and external stdlib, it's not easy to tell.

Reply:
I see

Reply:
Ha, I removed it but now I remember the issue. There is an ambiguity reported between `(==)(x::ReadOnly, y::AbstractVector)` and `(==)(x::AbstractVector, y::ReadOnly)`, one taken from the sysimg, and the other one from the package. Without resolving this stupid ambiguity, we can't turn on `Aqua.jl`'s ambiguity testing. IIRC, then this ambiguity is also returned by `detect_ambiguities`, the current ambiguity test.